vpu/generic/vpuGeneric/docker-compose.yml

```diff
@@ -0,0 +1,80 @@
+version: "3.9"
+services:
+  vpugeneric:
+    build:
+      context: .
+      dockerfile: Dockerfile
+    image: ${DOCKER_LOGIN}/vpugeneric:${TAG}
+    environment:
+      #- MACHINE=verdin-imx8mm
+      - MACHINE=verdin-imx8mp
```

Comment:
Originally, I had just the `- MACHINE` here but it seems that torizon user's environment is not available to the container. Is this by design?

Reply:
You mean, accessing this value `ARG MACHINE=verdin-imx8mp`, that was added on the `Dockerfile`, when the container is running?

Reply:
Or you mean that when you pass the environment variable through the `docker-compose.yml` it doesn't show on the `env` inside the container?

Reply:
In Torizon, we can pass `$MACHINE` to `docker run ` using `-e`, e.g.:
```
# w/o -eMACHINE
$ docker run torizon/debian:3-bookworm sh -c 'echo MACHINE=${MACHINE}'
MACHINE=
# w/ -eMACHINE
$ docker run -eMACHINE torizon/debian:3-bookworm sh -c 'echo MACHINE=${MACHINE}'
MACHINE=verdin-imx8mm
```

I imagined passing only `- MACHINE` would have the same effect in compose as described here: https://docs.docker.com/compose/compose-file/compose-file-v3/#environment. Probably I am missing something obvious.

Reply:
This seems to be the default behavior when docker-compose runs services on a remote host.

e.g consider following `docker-compose.yml` running on a Torizon device
```
torizon@verdin-imx8mm-06827785:~/temp$ cat docker-compose.yml
version: "3.9"
services:
  vpugeneric:
    image: torizon/debian:3-bookworm
    environment:
      - MACHINE
    command: sh -c "env && echo from command MACHINE=$${MACHINE}"
torizon@verdin-imx8mm-06827785:~/temp$ docker-compose up
[+] Running 1/0
 ✔ Container temp-vpugeneric-1  Created                                                                           0.0s 
Attaching to temp-vpugeneric-1
temp-vpugeneric-1  | HOSTNAME=81426f09f436
temp-vpugeneric-1  | HOME=/root
temp-vpugeneric-1  | MACHINE=verdin-imx8mm
temp-vpugeneric-1  | PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
temp-vpugeneric-1  | LC_ALL=C.UTF-8
temp-vpugeneric-1  | PWD=/
temp-vpugeneric-1  | from command MACHINE=verdin-imx8mm
temp-vpugeneric-1 exited with code 0
```

However if we use the same .yml in our sample, MACHINE shows as empty:
```
 *  Executing task: DOCKER_HOST=192.168.51.107:2375 DOCKER_LOGIN=192.168.51.100:5002 TAG=arm64 GPU=-vivante docker-compose -p torizon up vpugeneric 

Creating network "torizon_default" with the default driver
Creating torizon_vpugeneric_1 ... done
Attaching to torizon_vpugeneric_1
vpugeneric_1  | HOSTNAME=3fe25946ebe5
vpugeneric_1  | HOME=/root
vpugeneric_1  | PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
vpugeneric_1  | LC_ALL=C.UTF-8
vpugeneric_1  | PWD=/
vpugeneric_1  | from command MACHINE=
torizon_vpugeneric_1 exited with code 0
```